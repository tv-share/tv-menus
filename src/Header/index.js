import "./header.*";
import Header from "./Header";

export default Header;