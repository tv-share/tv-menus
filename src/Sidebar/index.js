import "./sidebar.*";
import Sidebar from "./Sidebar";

export default Sidebar;