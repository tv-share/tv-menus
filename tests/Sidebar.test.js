import React from 'react';
import SidebarExample from "../examples/SidebarExample";
import { Sidebar } from "../src/index";
import renderer from 'react-test-renderer';

it('renders not expandable', () => {
    const component = renderer.create(
        <SidebarExample />
    );
    expect(component.toJSON()).toMatchSnapshot();
});

function createNodeMock(element) {
    if (element.props.className.trim() === 'sidebar') {
      return {
        clientWidth: 200
      };
    }
    return null;
  }

it('renders expandable closed', () => {
    const component = renderer.create(
        <SidebarExample expandable />,
        {createNodeMock}
    );
    expect(component.toJSON()).toMatchSnapshot();
});

it('renders expandable opened', () => {
    const component = renderer.create(
        <SidebarExample expandable />,
        {createNodeMock}
    );
    component.root.findByType(Sidebar).instance.openAction();
    expect(component.toJSON()).toMatchSnapshot();
});