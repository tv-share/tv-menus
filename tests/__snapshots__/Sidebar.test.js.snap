// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders expandable closed 1`] = `
<div
  className="sidebar "
  style={
    Object {
      "width": "60px",
    }
  }
>
  <div
    className="menu-btn -closed"
    onClick={[Function]}
  >
    <div
      className="line1"
    />
    <div
      className="line2"
    />
    <div
      className="line3"
    />
  </div>
  <ul>
    <li
      className="item"
      onClick={[Function]}
    >
      <img
        className="icon"
        src="https://storage.googleapis.com/material-icons/external-assets/v4/icons/svg/ic_account_circle_white_24px.svg"
      />
      Usuários
    </li>
    <li
      className="item"
      onClick={[Function]}
    >
      <img
        className="icon"
        src="https://storage.googleapis.com/material-icons/external-assets/v4/icons/svg/ic_feedback_white_24px.svg"
      />
      Relatórios
    </li>
    <li
      className="item"
      onClick={[Function]}
    >
      <img
        className="icon"
        src="https://storage.googleapis.com/material-icons/external-assets/v4/icons/svg/ic_settings_white_24px.svg"
      />
      Configurações dos Bagulhos
    </li>
  </ul>
</div>
`;

exports[`renders expandable opened 1`] = `
<div
  className="sidebar "
  style={
    Object {
      "width": "202px",
    }
  }
>
  <div
    className="menu-btn -open"
    onClick={[Function]}
  >
    <div
      className="line1"
    />
    <div
      className="line2"
    />
    <div
      className="line3"
    />
  </div>
  <ul>
    <li
      className="item"
      onClick={[Function]}
    >
      <img
        className="icon"
        src="https://storage.googleapis.com/material-icons/external-assets/v4/icons/svg/ic_account_circle_white_24px.svg"
      />
      Usuários
    </li>
    <li
      className="item"
      onClick={[Function]}
    >
      <img
        className="icon"
        src="https://storage.googleapis.com/material-icons/external-assets/v4/icons/svg/ic_feedback_white_24px.svg"
      />
      Relatórios
    </li>
    <li
      className="item"
      onClick={[Function]}
    >
      <img
        className="icon"
        src="https://storage.googleapis.com/material-icons/external-assets/v4/icons/svg/ic_settings_white_24px.svg"
      />
      Configurações dos Bagulhos
    </li>
  </ul>
</div>
`;

exports[`renders not expandable 1`] = `
<div
  className="sidebar "
  style={
    Object {
      "width": null,
    }
  }
>
  <ul>
    <li
      className="item"
      onClick={[Function]}
    >
      <img
        className="icon"
        src="https://storage.googleapis.com/material-icons/external-assets/v4/icons/svg/ic_account_circle_white_24px.svg"
      />
      Usuários
    </li>
    <li
      className="item"
      onClick={[Function]}
    >
      <img
        className="icon"
        src="https://storage.googleapis.com/material-icons/external-assets/v4/icons/svg/ic_feedback_white_24px.svg"
      />
      Relatórios
    </li>
    <li
      className="item"
      onClick={[Function]}
    >
      <img
        className="icon"
        src="https://storage.googleapis.com/material-icons/external-assets/v4/icons/svg/ic_settings_white_24px.svg"
      />
      Configurações dos Bagulhos
    </li>
  </ul>
</div>
`;
